syntax = "proto3";

package library;

message Staff {
    int32 id = 1;
    string username = 2;
    string email = 3;
    optional string full_name = 4;
}

message RegisterRequest {
    string username = 1;
    string email = 2;
    string password = 3;
    optional string full_name = 4;
}

message LoginRequest {
    string email = 1;
    string password = 2;
}

message LoginResponse {
    string access_token = 1;
    string token_type = 2;
}

message GetCurrentUserRequest {}


service AuthService {
    rpc Register(RegisterRequest) returns (Staff);
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc GetCurrentUser(GetCurrentUserRequest) returns (Staff);
}